### PR TITLE
feat (coupons): handle reusable coupons

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

Handle reusable coupons

## Description

Currently, the same coupon can be applied multiple times on a customer. With new changes, we are able to control if coupon is reusable or not.

New checkbox is added on coupon form; default state is reusable = true

### CODE REVIEW

File: src/hooks/useCreateEditCoupon.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ gql`
     name
     amountCurrency
     code
+    reusable
```
Comment (ansmonjol) [inline src/hooks/useCreateEditCoupon.ts:32]: Is that change needed ? Seems `reusable` is not used in here

**Review by ansmonjol — APPROVED** [review]
Just one comment otherwise LGTM !

File: src/hooks/useCreateEditCoupon.ts (hunk 1 above)
Comment (lovrocolic) [reply src/hooks/useCreateEditCoupon.ts:32]: Hmm, If I remember correctly it was needed (this is edit coupon fragment) since reusable checkbox is also editable...

Comment (ansmonjol) [reply src/hooks/useCreateEditCoupon.ts:32]: ok probably an issue with the fragment usage in here, that was already done before you jumped into this file.

Will resolve, as a refactor can come later
